src/org/opendatakit/briefcase/util/PullFromServerException.java

```diff
@@ -22,6 +22,10 @@
 import org.opendatakit.briefcase.reused.BriefcaseException;
 
 public class PullFromServerException extends BriefcaseException {
+  public PullFromServerException(List<FormStatus> forms, Throwable cause) {
+    super("Failure pulling forms from server. FormIds: " + forms.stream().map(f -> f.getFormDefinition().getFormId()).collect(joining(", ")), cause);
```

Comment:
Hmm, haven’t I seen this first argument somewhere before? :-)

Reply:
I'm thinking about extracting that construction to its own private method? Would that be acceptable?